cloudinit/config/cc_reset_rmc.py

```diff
@@ -0,0 +1,138 @@
+- creation of the unique RSCT node id to every instance/virtual machine
+  and ensure once set, it isn't changed subsequently by cloud-init.
+  In order to do so, it restarts RSCT service.
+
+Prerequisite of using this module is to install RSCT packages.
+
+**Internal name:** ``cc_reset_rmc``
+
+**Module frequency:** per instance
+
+**Supported distros:** rhel, sles and ubuntu
+
+"""
+import os
+
+from cloudinit import log as logging
+from cloudinit.settings import PER_INSTANCE
+from cloudinit import util
+from cloudinit import subp
+
+frequency = PER_INSTANCE
+PIDOF = 'pidof'
+
+# RMCCTRL is expected to be in system PATH (/opt/rsct/bin)
+# The symlink for RMCCTRL and RECFGCT are
+# /usr/sbin/rsct/bin/rmcctrl and
+# /usr/sbin/rsct/install/bin/recfgct respectively.
+RSCT_PATH = '/opt/rsct/install/bin'
+RMCCTRL = 'rmcctrl'
+RECFGCT = 'recfgct'
+
+LOG = logging.getLogger(__name__)
+
+NODE_ID_FILE = '/etc/ct_node_id'
+
+
+def handle(name, _cfg, cloud, _log, _args):
+    # Ensuring node id has to be generated only once during first boot
+    if cloud.datasource.platform_type == 'none':
+        LOG.debug('Skipping creation of new ct_node_id node')
+        return
+
+    orig_path = os.environ.get('PATH')
+    try:
+        add_path(orig_path)
+
+        LOG.debug('Attempting to reset RMC.')
+
+        node_id_before = get_node_id()
+        LOG.debug('Node ID at beginning of module: %s', node_id_before)
+
+        # Stop the RMC subsystem and all resource managers so that we can make
+        # some changes to it
+        try:
+            subp.subp([RMCCTRL, '-z'])
+        except Exception:
+            util.logexc(LOG, 'Failed to stop the RMC subsystem.')
+            raise
+
+        reconfigure_rsct_subsystems()
+
+        node_id_after = get_node_id()
+        LOG.debug('Node ID at end of module: %s', node_id_after)
+
+        # Check if new node ID is generated or not
+        # by comparing old and new node ID
+        if node_id_after == node_id_before:
+            msg = 'New node ID did not get generated.'
+            LOG.error(msg)
+            raise Exception(msg)
+
+    finally:
+        if orig_path:
+            os.environ['PATH'] = orig_path
+        else:
+            del os.environ['PATH']
+
+
+def reconfigure_rsct_subsystems():
+    # Reconfigure the RSCT subsystems, which includes removing all RSCT data
+    # under the /var/ct directory, generating a new node ID, and making it
+    # appear as if the RSCT components were just installed
+    try:
+        out = subp.subp([RECFGCT])[0]
+        LOG.debug(out.strip())
+    except subp.ProcessExecutionError:
+        util.logexc(LOG, 'Failed to reconfigure the RSCT subsystems.')
+        raise
+
+
+def get_node_id():
+    try:
+        fp = util.load_file(NODE_ID_FILE)
+        node_id = fp.split('\n')[0]
+        return node_id
+    except Exception:
+        util.logexc(LOG, 'Failed to get node ID from file %s.' % NODE_ID_FILE)
+        raise
+
+
+def add_path(orig_path):
```

Comment:
we do need still need to adjust this.
the presence of 'cc_reset_rmc' in config_modules should not result in a stack trace and cloud-init failing on a system that does not have RSCT_PATH.


in handle, just add a simple:

if not os.path.isdir(RSCT_PATH):
    LOG.debug("module disabled, RSCT_PATH not present")
    return